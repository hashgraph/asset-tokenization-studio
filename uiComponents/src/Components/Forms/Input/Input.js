var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import { Input as ChakraInput, InputGroup as ChakraInputGroup, InputLeftElement as ChakraInputLeftElement, InputRightElement as ChakraInputRightElement, } from "@chakra-ui/input";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import React from "react";
import _merge from "lodash/merge";
import { Text } from "../../Foundations/Text";
import { Box as ChakraBox, Center, Flex as ChakraFlex, useBreakpointValue, } from "@chakra-ui/react";
import { PhosphorIcon, Weight } from "@/Components/Foundations/PhosphorIcon";
import { CheckCircle, Warning, X } from "@phosphor-icons/react";
import { IconButton } from "@Components/Interaction/IconButton";
export var inputPartsList = [
    "container",
    "addonLeft",
    "addonRight",
    "labelContainer",
    "label",
    "subLabel",
    "input",
    "errorIcon",
    "topDescription",
    "bottomDescription",
];
export var defaultInputSize = { base: "lg", lg: "md" };
export var Input = forwardRef(function (_a, ref) {
    var size = _a.size, variant = _a.variant, id = _a.id, AddonLeft = _a.addonLeft, AddonRight = _a.addonRight, isInvalid = _a.isInvalid, isDisabled = _a.isDisabled, isRequired = _a.isRequired, label = _a.label, subLabel = _a.subLabel, _b = _a.placeholder, placeholder = _b === void 0 ? " " : _b, _c = _a.showRequired, showRequired = _c === void 0 ? true : _c, isSuccess = _a.isSuccess, isClearable = _a.isClearable, onClear = _a.onClear, bottomDescription = _a.bottomDescription, topDescription = _a.topDescription, sx = _a.sx, props = __rest(_a, ["size", "variant", "id", "addonLeft", "addonRight", "isInvalid", "isDisabled", "isRequired", "label", "subLabel", "placeholder", "showRequired", "isSuccess", "isClearable", "onClear", "bottomDescription", "topDescription", "sx"]);
    var formControl = useChakraFormControlContext() || {};
    var invalid = isInvalid !== null && isInvalid !== void 0 ? isInvalid : formControl.isInvalid;
    var defaultSize = useBreakpointValue(defaultInputSize);
    var inputSize = size || defaultSize;
    var _d = useChakraMultiStyleConfig("Input", {
        size: inputSize,
        variant: variant,
        addonLeft: AddonLeft,
        addonRight: AddonRight,
        isInvalid: invalid,
        isSuccess: isSuccess,
        isDisabled: isDisabled,
        hasLabel: Boolean(label),
        isClearable: isClearable,
        onClear: onClear
    }), _e = _d.errorIcon, ThemeErrorIcon = _e === void 0 ? function () { return (React.createElement(InputIcon, { icon: React.createElement(PhosphorIcon, { as: Warning, variant: "error" }) })); } : _e, _f = _d.validIcon, ThemeValidIcon = _f === void 0 ? function () { return (React.createElement(InputIcon, { icon: React.createElement(PhosphorIcon, { as: CheckCircle }) })); } : _f, _g = _d.clearIconButton, ThemeClearIcon = _g === void 0 ? function () { return (React.createElement(InputIconButton, { "aria-label": "Clear", onMouseDown: onClear, icon: React.createElement(PhosphorIcon, { as: X, weight: Weight.Bold }) })); } : _g, styles = __rest(_d, ["errorIcon", "validIcon", "clearIconButton"]);
    var inputStyle = React.useMemo(function () { return _merge(styles.input, sx); }, [styles.input, sx]);
    return (React.createElement(ChakraBox, { as: "label", sx: styles.labelContainer, htmlFor: id, flex: 1, position: "relative", ref: ref },
        label &&
            (React.isValidElement(label) ? (React.createElement(ChakraBox, { sx: styles.label }, label)) : (React.createElement(Text, { as: "span", id: "label", display: "flex", sx: styles.label },
                label,
                showRequired && isRequired && "*"))),
        subLabel &&
            (React.isValidElement(subLabel) ? (React.createElement(ChakraBox, { sx: styles.subLabel }, subLabel)) : (React.createElement(Text, { as: "span", id: "subLabel", display: "flex", sx: styles.subLabel }, subLabel))),
        topDescription &&
            (React.isValidElement(topDescription) ? (React.createElement(ChakraBox, { sx: styles.topDescription }, topDescription)) : (React.createElement(Text, { sx: styles.topDescription }, topDescription))),
        React.createElement(ChakraInputGroup, { overflow: "hidden", sx: styles.container },
            AddonLeft && (React.createElement(ChakraInputLeftElement, { children: AddonLeft, height: "full", sx: styles.addonLeft })),
            React.createElement(ChakraInput, __assign({ size: inputSize, variant: variant, id: id, sx: inputStyle, isDisabled: isDisabled, isInvalid: invalid, placeholder: placeholder }, props)),
            (AddonRight || isInvalid || isSuccess || isClearable) && (React.createElement(ChakraInputRightElement, { pointerEvents: "all", zIndex: "auto", height: "full", sx: styles.addonRight },
                isClearable && (React.createElement(ChakraFlex, { "data-testid": "input-clear-icon-button" },
                    React.createElement(ThemeClearIcon, null))),
                AddonRight,
                isInvalid && (React.createElement(ChakraFlex, { "data-testid": "input-error-icon" }, React.createElement(ThemeErrorIcon, null))),
                isSuccess && (React.createElement(ChakraFlex, { "data-testid": "input-success-icon" }, React.createElement(ThemeValidIcon, null)))))),
        bottomDescription &&
            (React.isValidElement(bottomDescription) ? (React.createElement(ChakraBox, { sx: styles.bottomDescription }, bottomDescription)) : (React.createElement(Text, { sx: styles.bottomDescription }, bottomDescription)))));
});
export var InputIcon = function (_a) {
    var icon = _a.icon;
    var SizedIcon = React.cloneElement(icon, __assign({ size: "xxs", weight: Weight.Fill, sx: { zIndex: 1 } }, icon.props));
    return (React.createElement(Center, { w: 6, h: 6 }, SizedIcon));
};
export var InputIconButton = function (_a) {
    var icon = _a.icon, props = __rest(_a, ["icon"]);
    var SizedIcon = React.cloneElement(icon, __assign({ size: "xxs", weight: Weight.Fill }, icon.props));
    return (React.createElement(IconButton, __assign({ size: "xs", variant: "tertiary", sx: { color: "inherit" }, icon: SizedIcon }, props)));
};
