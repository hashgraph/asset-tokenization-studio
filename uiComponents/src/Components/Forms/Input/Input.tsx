import { useFormControlContext as useChakraFormControlContext } from "@chakra-ui/form-control";
import type { InputProps as ChakraInputProps } from "@chakra-ui/input";
import {
  Input as ChakraInput,
  InputGroup as ChakraInputGroup,
  InputLeftElement as ChakraInputLeftElement,
  InputRightElement as ChakraInputRightElement,
} from "@chakra-ui/input";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import React from "react";
import _merge from "lodash/merge";
import { Text } from "../../Foundations/Text";

import {
  Box as ChakraBox,
  Center,
  Flex as ChakraFlex,
  useBreakpointValue,
} from "@chakra-ui/react";
import { PhosphorIcon, Weight } from "@/Components/Foundations/PhosphorIcon";
import { CheckCircle, Warning, X } from "@phosphor-icons/react";
import type { IconButtonProps } from "@Components/Interaction/IconButton";
import { IconButton } from "@Components/Interaction/IconButton";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";

export const inputPartsList: Array<
  | "labelContainer"
  | "label"
  | "subLabel"
  | "addonLeft"
  | "addonRight"
  | "input"
  | "container"
  | "errorIcon"
  | "topDescription"
  | "bottomDescription"
> = [
  "container",
  "addonLeft",
  "addonRight",
  "labelContainer",
  "label",
  "subLabel",
  "input",
  "errorIcon",
  "topDescription",
  "bottomDescription",
];

type Parts = typeof inputPartsList;

export type InputConfigProps = {
  size: string;
  variant: ChakraInputProps["variant"];
  addonLeft?: React.ReactNode;
  addonRight?: React.ReactNode;
  isInvalid?: boolean;
  isSuccess?: boolean;
  isDisabled?: boolean;
  hasLabel?: boolean;
  hasSubLabel?: boolean;
  isClearable?: boolean;
  onClear?: React.MouseEventHandler<HTMLButtonElement>;
  bottomDescription?: string | React.ReactElement;
  topDescription?: string | React.ReactElement;
};
export interface InputThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        addonLeft,
        variant,
        isInvalid,
        isDisabled,
        hasSubLabel,
      }: InputConfigProps) => Partial<InputThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export interface InputProps extends ChakraInputProps {
  addonLeft?: React.ReactElement;
  addonRight?: React.ReactElement;
  size?: string;
  label?: string | React.ReactElement;
  subLabel?: string | React.ReactElement;
  showRequired?: boolean;
  isSuccess?: boolean;
  isClearable?: boolean;
  onClear?: React.MouseEventHandler<HTMLButtonElement>;
  bottomDescription?: string | React.ReactElement;
  topDescription?: string | React.ReactElement;
}

export interface InputIconButtonProps extends IconButtonProps {
  children?: React.ReactElement;
  icon: React.ReactElement;
}

export type InputThemeStyle = Record<Parts[number], ChakraSystemStyleObject> & {
  errorIcon: React.FunctionComponent;
  validIcon: React.FunctionComponent;
  clearIconButton: React.FunctionComponent;
};

export const defaultInputSize = { base: "lg", lg: "md" };
export const Input: ComponentWithAs<"input", InputProps> = forwardRef<
  InputProps,
  "input"
>(
  (
    {
      size,
      variant,
      id,
      addonLeft: AddonLeft,
      addonRight: AddonRight,
      isInvalid,
      isDisabled,
      isRequired,
      label,
      subLabel,
      placeholder = " ",
      showRequired = true,
      isSuccess,
      isClearable,
      onClear,
      bottomDescription,
      topDescription,
      sx,
      ...props
    }: InputProps,
    ref
  ) => {
    const formControl = useChakraFormControlContext() || {};
    const invalid = isInvalid ?? formControl.isInvalid;

    const defaultSize = useBreakpointValue(defaultInputSize);
    const inputSize = size || defaultSize;

    const {
      errorIcon: ThemeErrorIcon = () => (
        <InputIcon icon={<PhosphorIcon as={Warning} variant="error" />} />
      ),
      validIcon: ThemeValidIcon = () => (
        <InputIcon icon={<PhosphorIcon as={CheckCircle} />} />
      ),
      clearIconButton: ThemeClearIcon = () => (
        <InputIconButton
          aria-label="Clear"
          onMouseDown={onClear}
          icon={<PhosphorIcon as={X} weight={Weight.Bold} />}
        />
      ),
      ...styles
    } = useChakraMultiStyleConfig("Input", {
      size: inputSize,
      variant,
      addonLeft: AddonLeft,
      addonRight: AddonRight,
      isInvalid: invalid,
      isSuccess,
      isDisabled,
      hasLabel: Boolean(label),
      isClearable,
      onClear,
    }) as InputThemeStyle;

    const inputStyle = React.useMemo<ChakraSystemStyleObject>(
      () => _merge(styles.input, sx),
      [styles.input, sx]
    );

    return (
      <ChakraBox
        as="label"
        sx={styles.labelContainer}
        htmlFor={id}
        flex={1}
        position="relative"
        ref={ref}
      >
        {label &&
          (React.isValidElement(label) ? (
            <ChakraBox sx={styles.label}>{label}</ChakraBox>
          ) : (
            <Text as="span" id="label" display="flex" sx={styles.label}>
              {label}
              {showRequired && isRequired && "*"}
            </Text>
          ))}
        {subLabel &&
          (React.isValidElement(subLabel) ? (
            <ChakraBox sx={styles.subLabel}>{subLabel}</ChakraBox>
          ) : (
            <Text as="span" id="subLabel" display="flex" sx={styles.subLabel}>
              {subLabel}
            </Text>
          ))}
        {topDescription &&
          (React.isValidElement(topDescription) ? (
            <ChakraBox sx={styles.topDescription}>{topDescription}</ChakraBox>
          ) : (
            <Text sx={styles.topDescription}>{topDescription}</Text>
          ))}
        <ChakraInputGroup overflow="hidden" sx={styles.container}>
          {AddonLeft && (
            <ChakraInputLeftElement
              children={AddonLeft}
              height="full"
              sx={styles.addonLeft}
            />
          )}

          <ChakraInput
            size={inputSize}
            variant={variant}
            id={id}
            sx={inputStyle}
            isDisabled={isDisabled}
            isInvalid={invalid}
            placeholder={placeholder}
            {...props}
          />
          {(AddonRight || isInvalid || isSuccess || isClearable) && (
            <ChakraInputRightElement
              pointerEvents="all"
              zIndex="auto"
              height="full"
              sx={styles.addonRight}
            >
              {isClearable && (
                <ChakraFlex data-testid="input-clear-icon-button">
                  <ThemeClearIcon />
                </ChakraFlex>
              )}
              {AddonRight}
              {isInvalid && (
                <ChakraFlex data-testid="input-error-icon">
                  {<ThemeErrorIcon />}
                </ChakraFlex>
              )}
              {isSuccess && (
                <ChakraFlex data-testid="input-success-icon">
                  {<ThemeValidIcon />}
                </ChakraFlex>
              )}
            </ChakraInputRightElement>
          )}
        </ChakraInputGroup>
        {bottomDescription &&
          (React.isValidElement(bottomDescription) ? (
            <ChakraBox sx={styles.bottomDescription}>
              {bottomDescription}
            </ChakraBox>
          ) : (
            <Text sx={styles.bottomDescription}>{bottomDescription}</Text>
          ))}
      </ChakraBox>
    );
  }
);

export const InputIcon = ({ icon }: { icon: React.ReactElement }) => {
  const SizedIcon = React.cloneElement(icon, {
    size: "xxs",
    weight: Weight.Fill,
    sx: { zIndex: 1 },
    ...icon.props,
  });

  return (
    <Center w={6} h={6}>
      {SizedIcon}
    </Center>
  );
};

export const InputIconButton = ({ icon, ...props }: InputIconButtonProps) => {
  const SizedIcon = React.cloneElement(icon, {
    size: "xxs",
    weight: Weight.Fill,
    ...icon.props,
  });

  return (
    <IconButton
      size="xs"
      variant="tertiary"
      sx={{ color: "inherit" }}
      icon={SizedIcon}
      {...props}
    />
  );
};
