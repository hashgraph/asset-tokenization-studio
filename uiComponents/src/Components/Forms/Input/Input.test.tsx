import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { render } from "@/test-utils";
import type { RenderResult } from "@testing-library/react";
import { Bluetooth } from "@phosphor-icons/react";
import React from "react";
import type { InputProps } from "./Input";
import { InputIcon, InputIconButton, Input } from "./Input";

describe(`<Input />`, () => {
  const id = "test-input";
  const label = "label";
  const placeholder = "placeholder";
  const defaultProps = {
    id,
    label,
    placeholder,
  };
  const sizeMedium = { size: "md" };
  const sizeLarge = { size: "lg" };
  const addonLeft = {
    addonLeft: <InputIcon icon={<PhosphorIcon as={Bluetooth} />} />,
  };
  const addonRight = {
    addonRight: (
      <InputIconButton
        aria-label="Bluetooth"
        onClick={jest.fn()}
        icon={<PhosphorIcon as={Bluetooth} />}
      />
    ),
  };
  const isDisabled = { isDisabled: true };
  const isInvalid = { isInvalid: true };
  const outlineVariant = { variant: "outline" };
  const topDescription = { topDescription: "This is a description" };
  const bottomDescription = { bottomDescription: "This is a description" };
  const subLabel = { subLabel: "This is a sublabel" };

  const getLabelElement = (component: RenderResult) =>
    component.container.querySelector("span#label");

  const factoryComponent = (props: InputProps = defaultProps) =>
    render(<Input {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly when size is medium", () => {
    const component = factoryComponent({ ...defaultProps, ...sizeMedium });
    expect(component.asFragment()).toMatchSnapshot("Using medium size");
  });

  test("shows correctly when size is large", () => {
    const component = factoryComponent({ ...defaultProps, ...sizeLarge });
    const labelBlock = getLabelElement(component);
    expect(labelBlock).toHaveTextContent(label);
    expect(component.asFragment()).toMatchSnapshot("Using large size");
  });

  test("shows correctly when is disabled", () => {
    const component = factoryComponent({ ...defaultProps, ...isDisabled });
    expect(component.asFragment()).toMatchSnapshot("Using as disabled");
  });

  test("shows correctly when is invalid", () => {
    const component = factoryComponent({ ...defaultProps, ...isInvalid });
    expect(component.asFragment()).toMatchSnapshot("Using as invalid");
  });

  test("shows correctly when uses outline variant", () => {
    const component = factoryComponent({ ...defaultProps, ...outlineVariant });
    expect(component.asFragment()).toMatchSnapshot("Using outline variant");
  });

  test("shows correctly when uses addon left", () => {
    const component = factoryComponent({ ...defaultProps, ...addonLeft });
    expect(component.asFragment()).toMatchSnapshot("Using addon left");
  });

  test("shows correctly when uses addon right", () => {
    const component = factoryComponent({ ...defaultProps, ...addonRight });
    expect(component.asFragment()).toMatchSnapshot("Using addon right");
  });

  test("shows correctly when isSuccess", () => {
    const component = factoryComponent({ ...defaultProps, isSuccess: true });
    expect(component.asFragment()).toMatchSnapshot("Is Success");
  });

  test("shows correctly when isRequired", () => {
    const component = factoryComponent({ ...defaultProps, isRequired: true });
    expect(component.asFragment()).toMatchSnapshot("Is required");
  });

  test("shows correctly when uses addon right and input is invalid", () => {
    const component = factoryComponent({
      ...defaultProps,
      ...addonRight,
      ...isInvalid,
    });
    expect(component.asFragment()).toMatchSnapshot(
      "Using addon right and invalid"
    );
  });

  test("shows correctly with topDescription", () => {
    const component = factoryComponent({ ...defaultProps, ...topDescription });
    expect(
      component.getByText(topDescription.topDescription)
    ).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot("Top Description");
  });

  test("shows correctly with bottomDescription", () => {
    const component = factoryComponent({
      ...defaultProps,
      ...bottomDescription,
    });
    expect(
      component.getByText(bottomDescription.bottomDescription)
    ).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot("Bottom Description");
  });
  test("shows correctly with subLabel", () => {
    const component = factoryComponent({
      ...defaultProps,
      ...subLabel,
    });
    expect(component.getByText(subLabel.subLabel)).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot("Bottom Description");
  });
});
