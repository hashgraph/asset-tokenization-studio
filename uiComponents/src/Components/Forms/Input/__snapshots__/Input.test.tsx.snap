// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when is disabled: Using as disabled 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        disabled=""
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when is invalid: Using as invalid 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        aria-invalid="true"
        class="chakra-input css-ybk3ki"
        id="test-input"
        placeholder="placeholder"
      />
      <div
        class="chakra-input__right-element css-1akgnmh"
      >
        <div
          class="css-k008qs"
          data-testid="input-error-icon"
        >
          <div
            class="css-amv3mm"
          >
            <svg
              class="chakra-icon css-1j9mq94"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when isRequired: Is required 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label*
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when isSuccess: Is Success 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-1nxcjzo"
        id="test-input"
        placeholder="placeholder"
      />
      <div
        class="chakra-input__right-element css-1akgnmh"
      >
        <div
          class="css-k008qs"
          data-testid="input-success-icon"
        >
          <div
            class="css-amv3mm"
          >
            <svg
              class="chakra-icon css-3gomrt"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm45.66,85.66-56,56a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L112,148.69l50.34-50.35a8,8,0,0,1,11.32,11.32Z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when size is large: Using large size 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when size is medium: Using medium size 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-19hxx5h"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when uses addon left: Using addon left 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <div
        class="chakra-input__left-element css-irrpsr"
      >
        <div
          class="css-amv3mm"
        >
          <svg
            class="chakra-icon css-1bnjyrz"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192,176a8,8,0,0,1-3.2,6.4l-64,48A8,8,0,0,1,120,232a8,8,0,0,1-8-8V144L60.8,182.4a8,8,0,0,1-9.6-12.8L106.67,128,51.2,86.4a8,8,0,0,1,9.6-12.8L112,112V32a8,8,0,0,1,12.8-6.4l64,48a8,8,0,0,1,0,12.8L133.33,128l55.47,41.6A8,8,0,0,1,192,176Z"
            />
          </svg>
        </div>
      </div>
      <input
        class="chakra-input css-aaiujl"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when uses addon right and input is invalid: Using addon right and invalid 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        aria-invalid="true"
        class="chakra-input css-1rcvey7"
        id="test-input"
        placeholder="placeholder"
      />
      <div
        class="chakra-input__right-element css-1akgnmh"
      >
        <button
          aria-label="Bluetooth"
          class="chakra-button css-iw4qe6"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192,176a8,8,0,0,1-3.2,6.4l-64,48A8,8,0,0,1,120,232a8,8,0,0,1-8-8V144L60.8,182.4a8,8,0,0,1-9.6-12.8L106.67,128,51.2,86.4a8,8,0,0,1,9.6-12.8L112,112V32a8,8,0,0,1,12.8-6.4l64,48a8,8,0,0,1,0,12.8L133.33,128l55.47,41.6A8,8,0,0,1,192,176Z"
            />
          </svg>
        </button>
        <div
          class="css-k008qs"
          data-testid="input-error-icon"
        >
          <div
            class="css-amv3mm"
          >
            <svg
              class="chakra-icon css-1j9mq94"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when uses addon right: Using addon right 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-16qaat3"
        id="test-input"
        placeholder="placeholder"
      />
      <div
        class="chakra-input__right-element css-1akgnmh"
      >
        <button
          aria-label="Bluetooth"
          class="chakra-button css-iw4qe6"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-kpbtv0"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192,176a8,8,0,0,1-3.2,6.4l-64,48A8,8,0,0,1,120,232a8,8,0,0,1-8-8V144L60.8,182.4a8,8,0,0,1-9.6-12.8L106.67,128,51.2,86.4a8,8,0,0,1,9.6-12.8L112,112V32a8,8,0,0,1,12.8-6.4l64,48a8,8,0,0,1,0,12.8L133.33,128l55.47,41.6A8,8,0,0,1,192,176Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly when uses outline variant: Using outline variant 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly with bottomDescription: Bottom Description 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
    <p
      class="css-16fpc31"
    >
      This is a description
    </p>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly with subLabel: Bottom Description 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <span
      class="css-gwktuy"
      id="subLabel"
    >
      This is a sublabel
    </span>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Input /> shows correctly with topDescription: Top Description 1`] = `
<DocumentFragment>
  <label
    class="css-mimi7r"
    for="test-input"
  >
    <span
      class="css-1e9661m"
      id="label"
    >
      label
    </span>
    <p
      class="css-14yoljn"
    >
      This is a description
    </p>
    <div
      class="chakra-input__group css-pwmwmr"
      data-group="true"
    >
      <input
        class="chakra-input css-nkgdpn"
        id="test-input"
        placeholder="placeholder"
      />
    </div>
  </label>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
